linkerd/system/src/linux.rs

```diff
@@ -38,10 +41,15 @@ pub fn open_fds(pid: pid_t) -> io::Result<u64> {
     Ok(open)
 }
 
-pub fn max_fds() -> io::Result<Option<u64>> {
-    let limit = pid::limits_self()?.max_open_files;
-    let max_fds = limit.soft.or(limit.hard).map(|max| max as u64);
-    Ok(max_fds)
+pub fn max_fds() -> ProcResult<u64> {
+    let limits = Process::myself()?.limits()?.max_open_files;
+    match limits.soft_limit {
+        LimitValue::Unlimited => match limits.hard_limit {
+            LimitValue::Unlimited => Ok(0),
```

Comment:
What's 0 supposed to mean here? 🤔

Reply:
In order to retrieve sys_max we need to read the value from the file itself. Instead of doing that, I thought returning 0 here would be better as `hard_limit` would always be available.